lib/util/urltils.js

```diff
@@ -162,6 +162,31 @@ module.exports = {
     }
   },
 
+  /**
+   * Obfuscates path parameters with regex from config
+   *
+   * @param {Config} config The configuration containing the regex
+   * @param {string} path The path to be obfuscated
+   * @returns {string} The obfuscated path or the original path
+   */
+  obfuscatePath: function obfuscatePath(config, path) {
+    // TODO: handle SyntaxError: Invalid regular expression
+    if (typeof path !== 'string') {
+      return path
+    }
+
+    const { enabled, regex } = config.url_obfuscation
+    if (!enabled || !regex) {
+      return path
+    }
+
+    const { pattern, flags, replacement } = regex
+    const f = !flags ? '' : flags
+    const r = !replacement ? '' : replacement
+    const regexPattern = new RegExp(pattern, f)
+    return path.replace(regexPattern, `/${r}`)
```

Comment:
I'm unclear what this `/$` is doing with the replacement, can you explain this please?

Reply:
Somewhere internally (not sure where) the `/` is being used by the UI, see the example below:
<img width="984" alt="Screenshot 2023-02-03 at 10 58 56" src="https://user-images.githubusercontent.com/6328360/216586841-619f5f4b-efba-44ee-b98b-d1d8a22d83ca.png">

If there was no `/` the text in the UI would look like this:
```
jsonplaceholder.typicode.comusers/***
```

I can comment on this in the code if you think it's an important piece of info.

Reply:
wouldn't that just be part of your regex to decide if you want to replace the current `/` or not?

Reply:
i don't think we should include an extra `/`. customers can add that in their regex capture groups, etc